src/picture.rs

```diff
@@ -148,8 +151,72 @@ impl Default for Rav1dPicAllocator {
 
 unsafe extern "C" fn free_buffer(_data: *const u8, user_data: *mut c_void) {
     let pic_ctx: *mut pic_ctx_context = user_data as *mut pic_ctx_context;
-    (*pic_ctx).allocator.release_picture(&mut (*pic_ctx).pic);
-    free(pic_ctx as *mut c_void);
+    let pic_ctx = Box::from_raw(pic_ctx);
+    let data = &pic_ctx.pic.data;
+    pic_ctx
+        .allocator
+        .dealloc_picture_data(data.data, data.allocator_data);
+}
+
+impl Rav1dPicAllocator {
+    pub fn alloc_picture_data(
+        &self,
+        w: c_int,
+        h: c_int,
+        seq_hdr: Arc<DRav1d<Rav1dSequenceHeader, Dav1dSequenceHeader>>,
```

Comment:
Why not take an Option here? Then we don't need to unwrap and re-wrap.

Reply:
We're only wrapping here; the unwrap is done when calling this `fn`, and we need to unwrap to access `seq_hdr`'s fields.